src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HandshakeProtocol.cs

```diff
@@ -30,6 +30,23 @@ public static void WriteRequestMessage(HandshakeRequestMessage requestMessage, S
                 writer.WritePropertyName(ProtocolVersionName);
                 writer.WriteValue(requestMessage.Version);
                 writer.WriteEndObject();
+                writer.Flush();
+            }
+
+            TextMessageFormatter.WriteRecordSeparator(output);
+        }
+
+        public static void WriteResponseMessage(HandshakeResponseMessage responseMessage, IBufferWriter<byte> output)
+        {
+            using (var writer = CreateJsonTextWriter(output))
+            {
+                writer.WriteStartObject();
+                if (!string.IsNullOrEmpty(responseMessage.Error))
+                {
+                    writer.WritePropertyName(ErrorPropertyName);
+                    writer.WriteValue(responseMessage.Error);
+                }
+                writer.WriteEndObject();
```

Comment:
Missing `Flush`

Reply:
It isn't required because we're writing directly to the buffer